react/22/2.2/src/main.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { useState } from "react";
+
+function UserDetails({ data }) {
+  const [pass1, setPass1] = useState();
+  const [pass2, setPass2] = useState();
+  const [message, setMessage] = useState();
+
+  function register() {
+    if (pass1 && pass1 === pass2) {
+      setMessage("Successfully registered");
+    } else {
+      setMessage("Passwords do not match");
+    }
+  }
+
+  return (
+    <>
+      <table>
+        <tbody>
+          <tr>
+            <td>User name:</td>
+            <td>
+              <input type="text"></input>
+            </td>
+          </tr>
+          <tr>
+            <td>Password:</td>
+            <td>
+              <input
+                type="text"
+                value={pass1 || ""}
+                onChange={(e) => setPass1(e.target.value)}
+              ></input>
+            </td>
+          </tr>
+          <tr>
+            <td>Confirm password:</td>
+            <td>
+              <input
+                type="text"
+                value={pass2 || ""}
+                onChange={(e) => setPass2(e.target.value)}

```

Comment:
No need to separate update function for each field.
You can add a `name` field to the input and do this:
```js
const handleChange = event => {
  const name = event.target.name
  const value = event.target.value

  setForm(oldForm => ({ ...oldForm, [name]: value }))
}
```